planner/core/logical_plan_builder.go

```diff
@@ -2350,6 +2351,15 @@ func (b *PlanBuilder) buildDelete(delete *ast.DeleteStmt) (Plan, error) {
 		}
 	}
 
+	// Add a projection for the following case, otherwise the final schema will be the schema of the join.
+	// delete from t where a in (select ...) or b in (select ...)
+	if oldLen := oldSchema.Len(); oldLen != p.Schema().Len() {
+		proj := LogicalProjection{Exprs: expression.Column2Exprs(p.Schema().Columns[:oldLen])}.Init(b.ctx)
```

Comment:
it seems to delete target table isn't always the first table, MySQL support delete multiple tables syntax, although I never use that https://dev.mysql.com/doc/refman/5.7/en/delete.html :rofl:

Reply:
Yes, we should consider the case of multi-table deletion.

Reply:
Delete from A where ... // delete single table
Delete A B ... from ... // delete multiple table 
Delete from A B using ... // delete multiple table

Multiple tables case is more complex. This PR will only handle the single table cases.

Let's have a look what's the bug actually:
`select from A where A join B ...` there is a projection on A join B
but for 
`delete from A where A join B ...`, the projection is missing.

Reply:
What output should it be if it's a `multiple tables deleting`?